Client/Coordinators/Coordinator.swift

```diff
@@ -13,6 +13,9 @@ protocol Coordinator {
     func remove(child coordinator: Coordinator?)
 
     func handle(route: Route) -> Bool
+
+    @discardableResult
+    func find(route: Route) -> Coordinator?
```

Comment:
Small nit. I missed this in the PR that added this method but it could be better named. Find makes it seem like this function just checks some data and returns a value but this function actually handles the route also so it's a little. misleading. Not a blocker but if you do have some free time giving it a more descriptive name would be appreciated.